File: archives/20190519python/0225. Implement Stack using Queues.py
class MyStack:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.q=[]
    
    
    def qpush(self,q,x):
        q.insert(0,x)
        return
    def qpop(self,q):
        return q.pop()
    def qpeek(self,q):
        return q[-1]
    
    
    def push(self, x: int) -> None:
        """
        Push element x onto stack.
        """
        self.qpush(self.q,x)
            

    def pop(self) -> int:
        """
        Removes the element on top of the stack and returns that element.
        """
        tmp=[]
        while len(self.q)>1:
            self.qpush(tmp,self.qpop(self.q))
        rep=self.qpop(self.q)
        while tmp:
            self.qpush(self.q,self.qpop(tmp))
        return rep
        
    def top(self) -> int:
        """
        Get the top element.
        """
        tmp=[]
        while len(self.q)>1:
            self.qpush(tmp,self.qpop(self.q))
        rep=self.qpeek(self.q)
        self.qpush(tmp,self.qpop(self.q))
        while tmp:
            self.qpush(self.q,self.qpop(tmp))
        return rep

    def empty(self) -> bool:
        """
        Returns whether the stack is empty.
        """
        return self.q==[]
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: archives/C#/0225. Implement Stack using Queues.cs
public class MyStack {

    Queue<int> myStack= new Queue<int>();
    Queue<int> tmpQueue= new Queue<int>();
    /** Initialize your data structure here. */
    public MyStack() {
        
    }
    
    /** Push element x onto stack. */
    public void Push(int x) {
        myStack.Enqueue(x);
    }
    
    /** Removes the element on top of the stack and returns that element. */
    public int Pop() {
        while(myStack.Count>1){
            int x =myStack.Dequeue();
            tmpQueue.Enqueue(x);
        }
        int result=myStack.Dequeue();
        while(tmpQueue.Count>0){
            int x =tmpQueue.Dequeue();
            myStack.Enqueue(x);
        }
        return result;
    }
    
    /** Get the top element. */
    public int Top() {
        while(myStack.Count>1){
            int x =myStack.Dequeue();
            tmpQueue.Enqueue(x);
        }
        int result=myStack.Dequeue();
        tmpQueue.Enqueue(result);
        while(tmpQueue.Count>0){
            int x =tmpQueue.Dequeue();
            myStack.Enqueue(x);
        }
        return result;
    }
    
    /** Returns whether the stack is empty. */
    public bool Empty() {
        return myStack.Count==0;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.Push(x);
 * int param_2 = obj.Pop();
 * int param_3 = obj.Top();
 * bool param_4 = obj.Empty();
 */
